camera-viewfinder-compose/src/main/java/com/google/jetpackcamera/viewfinder/CameraPreview.kt

```diff
@@ -15,106 +15,202 @@
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.asExecutor
-import kotlinx.coroutines.flow.mapNotNull
 
-private const val TAG = "Preview"
+private const val TAG = "CameraPreview"
 
 @Composable
 fun CameraPreview(
     modifier: Modifier,
     implementationMode: ImplementationMode = ImplementationMode.COMPATIBLE,
     onSurfaceProviderReady: (SurfaceProvider) -> Unit = {},
-    onRequestBitmapReady: (() -> Bitmap?) -> Unit,
+    // TODO(b/267488554): Remove setSurfaceView after implementing an alternative to
+    //  metering point factory
     setSurfaceView: (View) -> Unit
 ) {
     Log.d(TAG, "CameraPreview")
 
     val surfaceRequest by produceState<SurfaceRequest?>(initialValue = null) {
         onSurfaceProviderReady(
             SurfaceProvider { request ->
+                Log.d(TAG, "newSurfaceRequest")
                 value?.willNotProvideSurface()
                 value = request
             }
         )
     }
 
-    PreviewSurface(
+    ViewFinder(
         modifier = modifier,
         surfaceRequest = surfaceRequest,
-        setView = setSurfaceView,
-        onRequestBitmapReady = onRequestBitmapReady,
         implementationMode = implementationMode
     )
 }
 
+@SuppressLint("RestrictedApi")
 @Composable
-fun PreviewSurface(
+fun ViewFinder(
     modifier: Modifier,
     surfaceRequest: SurfaceRequest?,
-    onRequestBitmapReady: (() -> Bitmap?) -> Unit,
-    implementationMode: ImplementationMode = ImplementationMode.COMPATIBLE,
-    setView: (View) -> Unit
+    implementationMode: ImplementationMode
 ) {
     Log.d(TAG, "PreviewSurface")
 
-    var surface: Surface? by remember { mutableStateOf(null) }
+    if (surfaceRequest != null) {
+        val transformationInfo by produceState<TransformationInfo?>(
+            key1 = surfaceRequest,
+            initialValue = null
+        ) {
+            surfaceRequest!!.setTransformationInfoListener(
+                Dispatchers.Main.asExecutor()
+            ) {
+                Log.d(TAG, "TransformationInfo: $it")
+                value = it
+            }
+        }
+
+        var surface by remember { mutableStateOf<Surface?>(null) }
+        var parentViewSize by remember { mutableStateOf(Size(1, 1)) }
 
-    LaunchedEffect(surfaceRequest, surface) {
-        Log.d(TAG, "LaunchedEffect")
-        snapshotFlow {
-            if (surfaceRequest == null || surface == null) {
-                null
-            } else {
-                Pair(surfaceRequest, surface)
+        Box(
+            modifier = modifier
+                .onGloballyPositioned {
+                    parentViewSize = Size(it.size.width, it.size.height)
+                }
+                .wrapContentSize(unbounded = true, align = Alignment.Center)
+        ) {
+            if (transformationInfo != null) {
+                TransformedSurface(
+                    surfaceRequest = surfaceRequest,
+                    transformationInfo = transformationInfo!!,
```

Comment:
I tried this in my local repo and it worked fine. This is a common pattern for variables that are retrieved through a getter. Instead of doing a null check, you explicitly capture the current non-null value in a block, and then refer to that value and not the original variable. The code looks like this:

```
        transformationInfo?.let {
            TransformedSurface(
                surfaceRequest = surfaceRequest,
                transformationInfo = it,
                parentViewSize = parentViewSize,
                implementationMode = implementationMode,
                onSurface = {
                    surface = it
                }
            )
        }
```